src/App.tsx

```diff
@@ -15,36 +23,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreperedGoods(
+  goods: Goods,
+  sortField: SortStatus,
+  reverseField: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (typeof sortField !== 'undefined') {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortStatus.SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+        case SortStatus.SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [reverseField, setReverseField] = useState(false);
+  const [sortField, setSortField] = useState<SortStatus>(
+    SortStatus.SORT_DEFAULT
+  );
+  const visibleGoods = getPreperedGoods(
+    goodsFromServer,
+    sortField,
+    reverseField,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortStatus.SORT_FIELD_ALPHABETICALLY)}
+          type="button"
+          className={`button is-info
+          ${sortField === SortStatus.SORT_FIELD_ALPHABETICALLY ? '' : 'is-light'}`}
```

Comment:
it is better to use libraries like `classnames` or `clsx` for conditional classnames